backend/common/utils/dl_sources/uri.py

```diff
@@ -160,6 +161,20 @@ def download(self, local_file_name: str):
         downloader.download(self.uri, local_file_name)
 
 
+class CXGPublicURL(S3URL):
```

Comment:
missing a validate abstractmethod implementation. I'm assuming we'll want to make sure we only accept datasets from the expected domain / format

Reply:
NVM, just noticed it extends S3URL.